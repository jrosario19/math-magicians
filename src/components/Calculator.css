body {
  display: flex;
  justify-content: center;
  align-items: flex-start;
  min-height: 98vh;
  background-color: #adc0db;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style: none;
  text-decoration: none;
}

.calculator {
  display: grid;
  grid-template-columns: repeat(4, 136px);
  grid-auto-rows: minmax(80px, auto);
  justify-content: flex-end;
}

.calculator-container {
  display: flex;
  margin-top: 85px;
  justify-content: space-between;
}

.calculator-headline {
  font-size: 35px;
}

.output {
  grid-column: 1 / 5;
  background-color: #858695;
  color: white;
  font-size: 2.5rem;
  text-align: right;
  border: none;
  padding-right: 10px;
  outline: none;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}

button {
  font-size: 1.25rem;
  border: 1px solid #e0e0e0;
}

button:hover {
  transform: scale(1.02);
  border: 1px solid blue;
}

#three-spanned {
  grid-column: 1 / 3;
}

.operator {
  background-color: #f5913e;
}
