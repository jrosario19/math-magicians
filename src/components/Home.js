import React from 'react';
import { LoremIpsum } from 'react-lorem-ipsum';

const Home = () => (
  <section className="section container">
    <h2 className="headline">Welcome to our page!</h2>
    <div className="section-text">
      {/* <LoremIpsum /> */}
      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus at eligendi explicabo asperiores
         similique molestiae totam assumenda doloribus, consectetur quod, quam officiis molestias tempore nemo, 
         ad nisi quia provident expedita.</p>
    </div>
    <div className="section-text">
      {/* <LoremIpsum /> */}
      <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Fuga, voluptatibus distinctio repellendus 
      asperiores maiores temporibus, harum provident qui dolores, error soluta tenetur hic labore modi 
      consequuntur id quod nam repudiandae?
      </p>
    </div>
  </section>
);

export default Home;
