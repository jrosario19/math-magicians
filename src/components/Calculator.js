import React, { useState } from 'react';
import Button from './Button';
import Display from './Display';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [inputValue, setInputValue] = useState(0);

  const onClickEventHandler = (e) => {
    const newValue = calculate(inputValue, e.target.innerText);
    setInputValue(newValue);
    const outPut = document.getElementById('output');
    if (newValue.next !== null) {
      outPut.textContent = newValue.next;
    } else {
      outPut.textContent = newValue.total;
    }
  };

  return (
    <div className="calculator-container container">
      <h2 className="calculator-headline container">Let&apos;s do some math!</h2>
      <div className="calculator container">
        <Display />
        <Button text="AC" onClick={onClickEventHandler} />
        <Button text="+/-" onClick={onClickEventHandler} />
        <Button text="%" onClick={onClickEventHandler} />
        <Button className="operator" text="&divide;" onClick={onClickEventHandler} />
        <Button text="7" onClick={onClickEventHandler} />
        <Button text="8" onClick={onClickEventHandler} />
        <Button text="9" onClick={onClickEventHandler} />
        <Button className="operator" text="x" onClick={onClickEventHandler} />
        <Button text="4" onClick={onClickEventHandler} />
        <Button text="5" onClick={onClickEventHandler} id="5"/>
        <Button text="6" onClick={onClickEventHandler} />
        <Button className="operator" text="-" onClick={onClickEventHandler} />
        <Button text="1" onClick={onClickEventHandler} />
        <Button text="2" onClick={onClickEventHandler} />
        <Button text="3" onClick={onClickEventHandler} />
        <Button className="operator" text="+" onClick={onClickEventHandler} />
        <Button id="three-spanned" text="0" onClick={onClickEventHandler} />
        <Button text="." onClick={onClickEventHandler} />
        <Button className="operator" text="=" onClick={onClickEventHandler} />
      </div>

    </div>
  );
};

export default Calculator;
