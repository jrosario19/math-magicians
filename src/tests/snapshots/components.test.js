import renderer from 'react-test-renderer';
import React from 'react';

import Home from '../../components/Home';
import Quote from '../../components/Quote';
import Calculator from '../../components/Calculator';

describe('Render of the webpage', () => {
  test('Render Home component', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render Quote component', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Render Calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
