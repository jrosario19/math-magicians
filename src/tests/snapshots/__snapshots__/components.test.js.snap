// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render of the webpage Render Calculator component 1`] = `
<div
  className="calculator-container container"
>
  <h2
    className="calculator-headline container"
  >
    Let's do some math!
  </h2>
  <div
    className="calculator container"
  >
    <div
      className="output"
    >
      <div
        id="output"
      >
        0
      </div>
    </div>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="operator"
      id=""
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="operator"
      id=""
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className=""
      id="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="operator"
      id=""
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="operator"
      id=""
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className=""
      id="three-spanned"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className=""
      id=""
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="operator"
      id=""
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Render of the webpage Render Home component 1`] = `
<section
  className="section container"
>
  <h2
    className="headline"
  >
    Welcome to our page!
  </h2>
  <div
    className="section-text"
  >
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus at eligendi explicabo asperiores similique molestiae totam assumenda doloribus, consectetur quod, quam officiis molestias tempore nemo, ad nisi quia provident expedita.
    </p>
  </div>
  <div
    className="section-text"
  >
    <p>
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Fuga, voluptatibus distinctio repellendus asperiores maiores temporibus, harum provident qui dolores, error soluta tenetur hic labore modi consequuntur id quod nam repudiandae?
    </p>
  </div>
</section>
`;

exports[`Render of the webpage Render Quote component 1`] = `
<div
  className="quote container"
>
  <p
    className="quote-text-container"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. William Paul Thurston
  </p>
</div>
`;
